### PR TITLE
Fix 11771 - cross-project references for projects using generative type providers

### PR DESCRIPTION
**Pull request by dsyme**
This fixes regression #11771

1. The regression was in the addition of these lines: https://github.com/dotnet/fsharp/pull/11485/files#diff-015b2a27e0e013656556b5d7e867b1f66a17e145606d4a9abea5eeadd8b7820cR1581

2. We had had this bug before, fixed here: https://github.com/dotnet/fsharp/pull/3236.  

3. An explicit test was added for it however it was disabled three years ago or so and never got re-enabled https://github.com/dotnet/fsharp/blame/main/tests/service/MultiProjectAnalysisTests.fs#L905

* [x] I have tested manually
* [x] I have re-enabled compiler service testing for projects referencing type providers (pending getting it green)
* [x] I have enabled the testing for .NET Core too (pending getting it green)

### CODE REVIEW

**Comment by dsyme** [comment]
@TIHan this is ready, pending green

**Comment by dsyme** [comment]
OK, ready

File: src/fsharp/service/FSharpCheckerResults.fs [hunk 1]
```diff
@@ -2184,7 +2184,7 @@ type FSharpCheckProjectResults
           keepAssemblyContents: bool,
           diagnostics: FSharpDiagnostic[],
           details:(TcGlobals * TcImports * CcuThunk * ModuleOrNamespaceType * Choice<IncrementalBuilder, TcSymbolUses> *
-                   TopAttribs option * IRawFSharpAssemblyData option * ILAssemblyRef *
+                   TopAttribs option * ILAssemblyRef *
```
Comment (TIHan) [inline src/fsharp/service/FSharpCheckerResults.fs:2187]: Yay! A dep removed :)

File: src/fsharp/service/service.fs [hunk 2]
```diff
@@ -279,9 +279,12 @@ type BackgroundCompiler(
...
                                 | Some ilReader ->
                                     let ilModuleDef, ilAsmRefs = ilReader.ILModuleDef, ilReader.ILAssemblyRefs
-                                    return RawFSharpAssemblyData(ilModuleDef, ilAsmRefs) :> IRawFSharpAssemblyData |> Some
+                                    let data = RawFSharpAssemblyData(ilModuleDef, ilAsmRefs) :> IRawFSharpAssemblyData
+                                    return ProjectAssemblyDataResult.Available data
                                 | _ ->
-                                    return None
+                                    // Note 'false' - if a PEReference doesn't find an ILModuleReader then we don't
+                                    // continue to try to use an on-disk DLL
+                                    return ProjectAssemblyDataResult.Unavailable false
```
Comment (TIHan) [inline src/fsharp/service/service.fs:287]: Ok, so this should behave the same as before. - We basically only do `useOnDisk` for type providers.

**Review by TIHan — APPROVED** [review]

File: src/fsharp/service/service.fs (hunk 2 above)
Comment (dsyme) [reply src/fsharp/service/service.fs:287]: Yes, though I suspect it's right for this path too - it's hard to imagine a situation where dropping the reference on the floor is better than reusing an on-disk DLL

**Pull request merged**